akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1013,11 +1012,7 @@ trait FlowOps[+Out, +Mat] {
    * '''Cancels when''' downstream cancels
    *
    */
-  def flatten[U](strategy: FlattenStrategy[Out, U]): Repr[U, Mat] = strategy match {
-    case scaladsl.FlattenStrategy.Concat | javadsl.FlattenStrategy.Concat ⇒ deprecatedAndThen(ConcatAll())
-    case _ ⇒
-      throw new IllegalArgumentException(s"Unsupported flattening strategy [${strategy.getClass.getName}]")
-  }
+  def flattenConcat[U]()(implicit ev: Out <:< Source[U, _]): Repr[U, Mat] = deprecatedAndThen(ConcatAll())
```

Comment:
@rkuhn Do you want me to take a stab at a flatConcat and remove flattenConcat now when we have the chance to do so? (even if the default implementation is map(f).andThen(ConcatAll)?)